### PR TITLE
enable a few ESLint rules

### PR DESCRIPTION
**Pull request by mrm007**
Rules enabled:

- [`@typescript-eslint/consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports/)
- [`import/no-duplicates`](https://github.com/import-js/eslint-plugin-import/blob/v2.26.0/docs/rules/no-duplicates.md)
- [`import/no-relative-packages`](https://github.com/import-js/eslint-plugin-import/blob/v2.26.0/docs/rules/no-relative-packages.md)

### CODE REVIEW

**Comment by mrm007** [comment]
Converting to draft while I tweak a few ESLint settings.

@seek-oss/braid-maintainers how do we feel about [@typescript-eslint/consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/)?

**Comment by michaeltaranto** [comment]
> Converting to draft while I tweak a few ESLint settings.
> 
> @seek-oss/braid-maintainers how do we feel about [@typescript-eslint/consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/)?

I quite like the explicitness. Does that disallow single-lined multi-typed imports? Eg.
```
import { fn, Foo } from 'Foo';
```
(where `fn` is a function and `Foo` is a type)

**Comment by mrm007** [comment]
@michaeltaranto 

The rule auto-fixes

```ts
import { fn, Foo } from 'foo';
```

to 

```ts
import type { Foo } from 'foo';
import { fn } from 'Foo';
```

But if you (re)write it as


```ts
import { fn, type Foo } from 'foo';
```

that's also valid

**Review by michaeltaranto — COMMENTED** [review]
Is the thinking that we will touch many of these files twice now? Once to extract type imports to standalone imports, and a second to revert to inline when the plugin supports that preference?

Outside of that, a few other minor comments.

File: package.json [hunk 1]
```diff
@@ -59,7 +59,8 @@
     "enquirer": "^2.3.6",
     "escape-string-regexp": "^4.0.0",
     "eslint": "^8.21.0",
-    "eslint-config-seek": "10.0.0",
+    "eslint-config-seek": "^10.1.0",
+    "eslint-plugin-canonical": "^3.1.0",
```
[the remark this answers is not in the document]
Comment (mrm007) [reply package.json:63]: If TypeScript 4.5 is a blocker, I would add it later; forgot to remove it now.

Same thing with adding it to `eslint-config-seek`, aside from being (too) opinionated. If TS 4.5 stops people from upgrading, then it's not great.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const fs = require('fs');
+
+const workspace = yaml.load(
+  // eslint-disable-next-line no-sync
+  fs.readFileSync(`${__dirname}/pnpm-workspace.yaml`, 'utf-8'),
+);
+
+module.exports = {
+  extends: '',
+  extends: 'seek',
```
[the remark this answers is not in the document]
Comment (mrm007) [reply .eslintrc.js:13]: Sloppy...

**Review by michaeltaranto — APPROVED** [review]
Looks good to me 😅

**Pull request merged**